server/src/main/scala/drt/server/feeds/stn/STNForecastXLSExtractor.scala

```diff
@@ -38,33 +37,44 @@ object STNForecastXLSExtractor {
 
     val lgwWorkSheet = workbook(xlsFilePath)
 
-    val sheet = sheetMapByIndex(3, lgwWorkSheet)
-    val arrivalRows: Seq[STNForecastFlightRow] = for {
+    val sheet = sheetMapByName("Arrivals by flight", lgwWorkSheet)
+
+    val headingIndexByNameMap: Map[String, Int] = headingIndexByName(sheet.getRow(1))
+
+    val arrivalRowsTry: Seq[Try[STNForecastFlightRow]] = for {
       rowNumber <- 2 to sheet.getLastRowNum
       row = sheet.getRow(rowNumber)
       if row.getCell(1) != null && row.getCell(1).getCellType != Cell.CELL_TYPE_BLANK
     } yield {
-      val scheduledCell = numericCellOption(1, row).getOrElse(0.0)
-      val carrierCodeCell = stringCellOption(2, row).getOrElse("")
-      val flightNumberCell = stringCellOption(3, row).getOrElse("")
-      val originCell = stringCellOption(4, row)
-      val maxPaxCell = numericCellOption(5, row).getOrElse(0.0)
-      val totalCell = numericCellOption(6, row).getOrElse(0.0)
-      val internationalDomesticCell = stringCellOption(10, row)
-      val scheduled = SDate(DateUtil.getJavaDate(scheduledCell, TimeZone.getTimeZone("UTC")).getTime)
-
-      STNForecastFlightRow(scheduledDate = scheduled,
-        flightCode = s"$carrierCodeCell$flightNumberCell",
-        origin = originCell.getOrElse(""),
-        internationalDomestic = internationalDomesticCell.getOrElse(""),
-        totalPax = totalCell.toInt,
-        maxPax = maxPaxCell.toInt
-      )
+      Try {
+        val scheduledCell = numericCellOption(headingIndexByNameMap("SCHEDULED TIME& DATE"), row).getOrElse(0.0)
+        val carrierCodeCell = stringCellOption(headingIndexByNameMap("AIRLINE"), row).getOrElse("")
+        val flightNumberCell = stringCellOption(headingIndexByNameMap("FLIGHT NUMBER"), row).getOrElse("")
+        val originCell = stringCellOption(headingIndexByNameMap("DESTINATION / ORIGIN"), row)
+        val maxPaxCell = numericCellOption(headingIndexByNameMap("FLIGHT CAPACITY"), row).getOrElse(0.0)
+        val totalCell = numericCellOption(headingIndexByNameMap("FLIGHT FORECAST"), row).getOrElse(0.0)
+        val internationalDomesticCell = stringCellOption(headingIndexByNameMap("TYPE"), row)
+        val scheduled = SDate(DateUtil.getJavaDate(scheduledCell, TimeZone.getTimeZone("UTC")).getTime)
+
+        STNForecastFlightRow(scheduledDate = scheduled,
+          flightCode = s"$carrierCodeCell$flightNumberCell",
+          origin = originCell.getOrElse(""),
+          internationalDomestic = internationalDomesticCell.getOrElse(""),
+          totalPax = totalCell.toInt,
+          maxPax = maxPaxCell.toInt
+        )
+      }
     }
 
+    val arrivalRows = arrivalRowsTry.toList.flatMap {
+      case Success(a) => Some(a)
+      case Failure(e) => log.error(s"Invalid data ${e.getMessage}")
```

Comment:
we should maybe make this a warning, seeing as it's recoverable not not very consequential.

Reply:
Fine with it .